tethysapp/hydro_var_monitor/controllers.py

```diff
@@ -215,21 +292,70 @@ def get_plot(request):
     return JsonResponse(json.loads(json.dumps(plot_data)))
 
 
-@controller(name='get_predefined', url='hydro-var-monitor/get-predefined',app_workspace=True)
+@controller(name='admin-upload-exact-zipped-data', url='hydro-var-monitor/admin/unzip-exact', app_workspace=True)
+def unzip_exact(request, app_workspace):
+    try:
+        workspace_path = app_workspace.path
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            for chunk in request.FILES['exact-json'].chunks():
+                f.write(chunk)
+
+        target_directory = os.path.join(workspace_path, 'Exact')
+        with zipfile.ZipFile(f.name, 'r') as zip_ref:
+            zip_ref.extractall(target_directory)
+        # for file in files:
+        #   new_observation_path = os.path.join(workspace_path, 'observations', files[file].name)
+        #  with open(new_observation_path, 'wb') as dst:
+        #      for chunk in files[file].chunks():
+        #         dst.write(chunk)
+
+        # step 2
+        # read the names of directories and the contents of each directory to make a datastructure that looks like
+        # include in context so it can be used to auto populate choices
+        #  and then save it to the workspace as a json file
+
+        # return also options_for_each_directory
+        # return {'success': True, 'data': {}}
+        return
+
+    except Exception as e:
+        return {'success': False, 'error': f'Error Processing Request: {e}'}
+
+
+@controller(name='admin-upload-simplified-zipped-data', url='hydro-var-monitor/admin/unzip-simplified',
```

Comment:
this should require the user to be logged in and be admin. See tethys docs for how to use the @login_required decorator and another decorator for being and admin user which i don't remember the name of.